middleware/authentication.js

```diff
@@ -0,0 +1,23 @@
+const User = require('../models/User')
+const jwt = require('jsonwebtoken')
+const { UnauthenticatedError } = require('../errors')
+
+const auth = async (req, res, next) => {
+  // check header
+  const authHeader = req.headers.authorization
+  if (!authHeader || !authHeader.startsWith('Bearer')) {
```

Comment:
[Requesting Additional Feature]
OK - so this is what we did in class with Bearer token authentication. And that's good, but we've actually sent a Cookie in the login request which is easier for the front-end team to use securely than Bearer Token authentication is.

I want us to support both. Bearer token for API requests from say Postman or Pupeteer where ease of use would be more important. Cookies for the front-end where being secure by default (without the front-end team having to worry about it constantly) is important.

Take a look at this blog post and see if you can't make both work
https://dev.to/franciscomendes10866/using-cookies-with-jwt-in-node-js-8fn

Reply:
I believe I resolved this issue using the recommended techniques in the article, and tested our login as well as register routes which seem to be functioning using the new authentication method